lib/src/rules/null_closures.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+library linter.src.rules.null_closures;
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:linter/src/analyzer.dart';
+import 'package:linter/src/util/dart_type_utilities.dart';
+
+const _desc = r'Do not pass `null` as an argument where a closure is expected.';
+
+const _details = r'''
```

Comment:
You should probably note in the description that this lint only catches explicit uses of null literals. It won't catch
```
var test = null;
[].firstWhere(test);
```
nor even
```
[].firstWhere((null));
```